ansible/roles/create-vm/tasks/main.yml

```diff
@@ -1,30 +1,46 @@
 ---
 
+#
+# Exclusion precedence:
+#   1. ppc64le is the only platform allowed to run eBPF
+#   2. specific exclusions defined group_vars/all.yml
+#
+# Consider each set_fact to be a rule, and subsequent rules are only
+# processed if the previous rules do not exclude this combination of
+# vm and collection method.
+#
+- name: Check eBPF excluded
+  set_fact:
+    excluded: "{{ test_platform != 'rhel-ppc64le' and vm_collection_method == 'ebpf' }}"
+
 - name: Check family excluded
   set_fact:
     excluded: "{{ [vm_family, vm_collection_method] in excluded_pairs }}"
-  when: vm_image is not defined
+  when: vm_image is not defined and excluded == false
 
 - name: Check image excluded
   set_fact:
     excluded: "{{ [vm_image, vm_collection_method] in excluded_pairs }}"
-  when: vm_image is defined
-
-- name: Create IBM Z VM
-  include_tasks: create-s390x-vm.yml
-  vars:
-    vsi_name: "{{ vm_name }}"
-  when: vm_arch == "s390x" and excluded == false
-
-- name: Create IBM POWER VM
-  include_tasks: create-ppc64le-vm.yml
-  vars:
-    pi_name: "{{ vm_name }}"
-  when: vm_arch == "ppc64le" and excluded == false
-
-- name: Create GCP VM
-  include_tasks: create-gcp-vm.yml
-  when: (vm_arch == "amd64" or vm_arch == "arm64") and excluded == false
+  when: vm_image is defined and excluded == false
+
+- block:
+  - name: Create IBM Z VM
+    include_tasks: create-s390x-vm.yml
+    vars:
+      vsi_name: "{{ vm_name }}"
+    when: vm_arch == "s390x"
+
+  - name: Create IBM POWER VM
+    include_tasks: create-ppc64le-vm.yml
+    vars:
+      pi_name: "{{ vm_name }}"
+    when: vm_arch == "ppc64le"
+
+  - name: Create GCP VM
+    include_tasks: create-gcp-vm.yml
+    when: (vm_arch == "amd64" or vm_arch == "arm64")
+
+  when: excluded == false
```

Comment:
I'm a bit concerned about the fact that with this approach to answer a question "what is excluded from the tests" one have to run the pipeline (physically or virtually in their head) to answer. Is there a way to structure this in a way that will make answering this question trivial?

Reply:
yeah that's a fair point. We could extend the current `excluded_pairs` variable to define entire platforms (e.g. `- [ 'rhel', 'ebpf' ]`) so then everything is centralised in the group_vars/all.yml file, and the first check here matches the others - `excluded: {{ [test_platform, vm_collection_method] in excluded_pairs }}` 

Ideally we'd want to sort of reverse the mechanism slightly for eBPF so power can opt-in, rather than having everything else opt-out, but that might require much more extensive changes to the way `excluded_pairs` works.

Hopefully we'll get an answer in the next couple of days about whether we can just drop eBPF altogether, at which point we can simplify back to how excluded_pairs worked to begin with, and in fact we could even drop the collection method and just exclude families or images

Reply:
for reference, this is what I mean https://github.com/stackrox/collector/pull/1677/commits/be8138db6fc6b9cbb368739fe7bd052e3d1517c0